### PR TITLE
gas optimization to MerkleMultiProof

### PR DESCRIPTION
**Pull request by nuel-ikwuoma**
added some gas optimization specific to EVM to MerkleMultiProof.sol
these changes should reduce the gas cost for calling the verifyProof routine

### CODE REVIEW

File: lib/MerkleMultiProof.sol [hunk 1]
```diff
@@ -8,7 +8,7 @@ struct Node {
 
 library MerkleMultiProof {
     function verifyProof(bytes32 root, Node[][] memory proof, Node[] memory leaves)
-        public
+        external
```
Comment (samparsky) [inline lib/MerkleMultiProof.sol:11]: change this to internal. To make the tests pass create a `contract` instead that uses the library

**Review by samparsky — CHANGES_REQUESTED** [review]

**Review by seunlanlege — APPROVED** [review]
Lgtm

**Comment by seunlanlege** [comment]
@nuel-ikwuoma you'll need to implement commit signing on your local git config & sign your commits. See https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits

**Comment by seunlanlege** [comment]
Then force push the newly signed commits

**Comment by nuel-ikwuoma** [comment]
@seunlanlege commit is verified

**Comment by seunlanlege** [comment]
> @seunlanlege commit is verified

All commits have to be verified

**Comment by seunlanlege** [comment]
Tested locally and saw some gas reductions, nothing too major.

**Pull request merged**